File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

int digits(int num);

int main(void)
{
    long long num;
    do
    {
        num = get_long("Number: ");
    }
    while (num < 0);

    int length = (int) (log(num) / log(10)) + 1, checksum = 0;
    int card[length];

    for(int i = 0; i < length; i++)
    {
        card[i] = (num % 10);
        num /= 10;
    }

    for (int i = 0; i < length; i++)
    {
        if(i % 2)
        {
            checksum += digits(2 * card[i]);
        }
        else
        {
            checksum += card[i];
        }
    }

    if(checksum % 10 != 0 || length < 13 || length > 16)
    {
        printf("INVALID\n");
    }
    else if(card[length - 1] == 4)
    {
        printf("VISA\n");
    }
    else if(card[length - 1] == 5 && (card[length - 2] == 1 || card[length - 2] == 2 || card[length - 2] == 3 || card[length - 2] == 4 || card[length - 2] == 5))
    {
        printf("MASTERCARD\n");
    }
    else if(card[length - 1] == 3 && (card[length - 2] == 4 || card[length - 2] == 7))
    {
        printf("AMEX\n");
    }
    else
    {
        printf("INVALID\n");
    }
}

int digits(int num)
{
    if(num < 10)
    {
        return num;
    }
    return digits(num / 10) + num % 10;
}

File: pset6/credit/credit.py
from cs50 import get_int


def main():
    n = str(get_int("Number: "))
    s, s1, c = 0, 0, 0

    for i in range(len(n) - 1, -1, -1):
        if (c % 2 == 1):
            s = s + digits((ord(n[i]) - ord('0')) * 2)
        else:
            s1 = s1 + (ord(n[i]) - ord('0'))
        c = c + 1

    if(((s + s1) % 10 != 0) or len(n) < 13 or len(n) > 16):
        print("INVALID")
    elif(n[0] == '4'):
        print("VISA")
    elif(n[0] == '5' and (n[1] in ['1', '2', '3', '4', '5'])):
        print("MASTERCARD")
    elif(n[0] == '3' and (n[1] == '4' or n[1] == '7')):
        print("AMEX")
    else:
        print("INVALID")


def digits(s):
    if(s < 10):
        return int(s)
    return int(s % 10) + int(digits(s / 10))


if __name__ == "__main__":
    main()